sources/engine/Stride.Rendering/Rendering/Mesh.cs

```diff
@@ -76,5 +81,133 @@ public Mesh(MeshDraw meshDraw, ParameterCollection parameters)
+        {
+            var shape = Shapes?.Keys?.Where(c => c.Name == shapeName).FirstOrDefault();
+            if (shape != null)
+            {
+                SetBendShapeWeight(shape, weight);
+            }
+        }
+
+        public void SetBlendShapeWeightByIndex(int index, float weight)
+        {
+            var shape = Shapes?.ElementAt(index);
+            if (shape != null)
+            {
+                SetBendShapeWeight(shape.Value.Key, weight);
+            }
+        }
+
+        void SetBendShapeWeight(Shape shape, float weight)
+        {
+            if (shape == null) { return; }
+            bool containsShape = Shapes != null && Shapes.ContainsKey(shape);
+            if (containsShape)
+            {
+                if (Shapes[shape] != weight)
+                {
+                    Shapes[shape] = weight;
+                    OnWeightChanged();
+                }
+            }
+        }
+             
+        public void UpdateBlendShapeImpact()
+        {
+            if(!IsVertexDataUpdated2BlendShapeData)
+            { 
+                ProcessBlendShapes();
+            }
+            else if (!IsVertexDataUpdated2BlendShapeWeight)
+            {
+                ProcessBlendShapes();
+            }
+            IsVertexDataUpdated2BlendShapeData = true;
+            IsVertexDataUpdated2BlendShapeWeight = true;
+        }
+
+   
+        /// <summary>
+        /// CORE BLENDSHAPE PROCESSING, CAN BE EXPENSIVE THEREFORE DONE A BLEND SHAPE IS ADDED THROUGH A FILE IMPORT OR PROGRAMATICALLY, OR THE SHAPE DEFINITION UPDATES PROGRAMATICALLY
+        /// 1. Update BDATAMAT to default
+        /// 2.VISIT EACH BLENDHSHAPE UPDATE TO NET IMPACT on vertex =( WEIGHT * NEXT IMPACT)
+        /// </summary>
+        unsafe void ProcessBlendShapes()
+        {
+            fixed (byte* bp = Draw.VertexData)
+            {
+                int _counter = 0;
+                for (int i = 0; i < Draw.VertexCount; i++)
+                {
+                    Vector3 netImpact = Vector3.Zero;
+                    foreach (var shape in Shapes)
+                    {
+                        if (shape.Key.VertexImpactedByBlendShapes.Contains(i))
+                        {
+                            ++_counter;
+                            var position = shape.Key.Position;
+                            var impact = new Vector3(position[i].X - Draw.VerticesOriginal[i].X, position[i].Y - Draw.VerticesOriginal[i].Y, position[i].Z - Draw.VerticesOriginal[i].Z);
+                            var impactWeighted = impact * shape.Value;
+                            netImpact += impactWeighted;
+                        }
+                    }
+
+                    Vector3* v = (Vector3*)(bp+ i*56);
+                    v->X = Draw.VerticesOriginal[i].X + netImpact.X;
+                    v->Y = Draw.VerticesOriginal[i].Y + netImpact.Y;
+                    v->Z = Draw.VerticesOriginal[i].Z + netImpact.Z;
+                }
```

Comment:
This should be done on the gpu as explained here https://github.com/stride3d/stride/issues/339#issuecomment-499306725
In method#1 Every shape with `shape.Value > 0` is uploaded to the GPU. `VertexImpactedByBlendShapes` doesn't exist, blendshape positions and vertex positions have the same length and map 1:1, e.g.: `blendshape[k]` is the position for `vertex[k]` if the weight of the blend shape is `1`, so indices that are not impacted by the blendshape are just set to the vertex's original position.

`Vector3* v = (Vector3*)(bp+ i*56);` this is a huge hack, the vertex buffer could have any kind of layout, so you would have to query the layout to find out where and how positions are stored. The gpu solution wouldn't have the same problem, so it doesn't matter ultimately.